components/cluster.py

```diff
@@ -230,11 +292,45 @@ def get_plot(graph, positions):
    """
     Xe = []
     Ye = []
+
+    X_foreshocks = []
+    Y_foreshocks = []
+    Hover_foreshocks = []
+
+    X_aftershocks = []
+    Y_aftershocks = []
+    Hover_aftershocks = []
+
+    max_magnitude = -1000
+    max_time = 0.0
     for n in graph:
         # we convert the time stamp to a human readeable format
-        Xe.append(dt.fromtimestamp(positions[n][0]//10**9))
+        mag = positions[n][1]
+        Xe.append(positions[n][0])
         Ye.append(positions[n][1])
 
+        if mag >= max_magnitude:
+            max_magnitude = mag
+            max_time = positions[n][0]
+
+    for n in graph:
+        mag = positions[n][1]
+        if mag == max_magnitude:
+            # cont
+            continue
+        text = "Pos:({},{})".format(positions[n][2], positions[n][3])
+        if positions[n][0] < max_time:
+            # foreshock
+            X_foreshocks.append(positions[n][0])
+            Y_foreshocks.append(positions[n][1])
+            Hover_foreshocks.append(text)
```

Comment:
Is there a reason the hover text should be different for the main shock and pre/aftershocks? It might be nice to see the same information for each, possibly containing all three features used for clustering. And instead of writing "Pos" in the hover info you could use location, which is maybe a bit clearer

Reply:
There is no particular reason other than maybe keep the components for each figure . The features used for the clustering are displayed, in the y-axis you have the magnitude and the hover information is the location.

Reply:
Looks good now!